src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -402,6 +402,7 @@ private void LowerConstantValueDecision(DecisionTree.ByValue byValue)
                     {
                         var loweredRight = kv.Key;
                         var loweredLeft = kv.Value;
+                        loweredRight = _factory.Convert(loweredLeft.Type, loweredRight);
```

Comment:
I've added some tests for additional conversions, including the requested conversions whose correctness were not affected by the PR.

@dotnet/roslyn-compiler May I please have a couple of reviews of this change?